modules/nextflow/src/main/groovy/nextflow/executor/BashWrapperBuilder.groovy

```diff
@@ -728,22 +733,17 @@ class BashWrapperBuilder {
                 builder.addEnv(var)
         }
 
-        // set up run docker params
-        builder.params(containerConfig)
-
-        // extra rule for the 'auto' temp dir temp dir
-        def temp = containerConfig.temp?.toString()
-        if( temp == 'auto' || temp == 'true' ) {
+        // extra rule for the 'auto' temp dir
+        if( containerConfig.getTemp() == 'auto' )
             builder.setTemp( changeDir ? '$NXF_SCRATCH' : '$(nxf_mktemp)' )
-        }
 
-        if( containerConfig.containsKey('kill') )
-            builder.params(kill: containerConfig.kill)
+        if( containerConfig.getKill() != null )
+            builder.params(kill: containerConfig.getKill())
 
-        if( containerConfig.writableInputMounts==false )
+        if( containerConfig.writableInputMounts()==false )
             builder.params(readOnlyInputs: true)
```

Comment:
It looks like some container scopes can specify `writableInputMounts = false` to mount folders into the container as read-only. There are tests for this option for apptainer, charliecloud, docker, podman, and singularity... so basically all of them.

If we want to keep this behavior, I suggest we add config option for each container scope where it's relevant. Or we could add an environment var.

Reply:
I think I will add explicit config options for `writableInputMounts` just for docker and charliecloud for now, and if people request it for other containers we can add it